isis/src/base/apps/framestitch/framestitch.cpp

```diff
@@ -0,0 +1,220 @@
+      return 0;
+    }
+
+    // Use the median to handle any outliers
+    return frameSizes.Median();
+  }
+
+  /**
+   * Compute the stats for an ISIS cube. This is the programmatic interface to
+   * the ISIS stats application.
+   *
+   * @param ui The User Interface to parse the parameters from
+   */
+  void framestitch(UserInterface &ui) {
+    ProcessByBrick process;
+
+    // It is very important that the even cube gets added first as later on
+    // we'll use frameNumber % 2 to index the input cube vector
+    QString evenCubeFile = ui.GetCubeName("EVEN");
+    Cube* evenCube = process.SetInputCube(evenCubeFile,
+                                          CubeAttributeInput(evenCubeFile));
+    QString oddCubeFile = ui.GetCubeName("ODD");
+    Cube* oddCube = process.SetInputCube(oddCubeFile,
+                                         CubeAttributeInput(oddCubeFile));
+
+
+    // Check that all the inputs are valid
+    if (evenCube->sampleCount() != oddCube->sampleCount() ||
+        evenCube->lineCount() != oddCube->lineCount() ||
+        evenCube->bandCount() != oddCube->bandCount()) {
+      QString msg = "Even and odd cube dimensions must match.";
+      throw IException(IException::User, msg, _FILEINFO_);
+    }
+
+    bool inputFlipped = false;
+
+    if (evenCube->hasGroup("Instrument") && oddCube->hasGroup("Instrument")) {
+      const PvlGroup &evenInst = evenCube->group("Instrument");
+      const PvlGroup &oddInst = oddCube->group("Instrument");
+      // Use the start time as an indicator of being the same original image
+      if (QString::compare(QString(evenInst["StartTime"]), QString(oddInst["StartTime"])) != 0) {
+        QString msg = "Even and odd cube start times must match.";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
+
+      if (evenInst.hasKeyword("DataFlipped") && oddInst.hasKeyword("DataFlipped")) {
+        if (toBool(evenInst["DataFlipped"]) != toBool(oddInst["DataFlipped"])) {
+          QString msg = "Both input cubes must be flipped or not flipped. Cannot combine "
+                        "a flipped and unflipped cube.";
+          throw IException(IException::User, msg, _FILEINFO_);
+        }
+        inputFlipped = toBool(evenInst["DataFlipped"]);
+      }
+    }
+
+    QString outCubeFile = ui.GetCubeName("TO");
+    Cube *outCube = process.SetOutputCube(
+          outCubeFile, CubeAttributeOutput(outCubeFile),
+          evenCube->sampleCount(), evenCube->lineCount(), evenCube->bandCount());
+
+    int frameSize = 1;
+    if (ui.WasEntered("FRAMESIZE")) {
+      frameSize = ui.GetInteger("FRAMESIZE");
+    }
+    else {
+      // User didn't pass the size of the frames so attempt to infer it from
+      // the cubes
+      int evenFrameSize = computeFrameSize(evenCube);
+      int oddFrameSize = computeFrameSize(oddCube);
+
+      if (evenFrameSize == 0) {
+        QString msg = "Failed to find any NULL frames in even cube when attempting "
```

Comment:
I think this should error message should say "non-NULL" rather than "NULL". Or even better, maybe, "failed to find any valid data in even cube".